pallets/investments/src/benchmarking.rs

```diff
@@ -10,3 +10,134 @@
 // but WITHOUT ANY WARRANTY; without even the implied warranty of
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
+
+use cfg_traits::{
+	benchmarking::{InvestmentIdBenchmarkHelper, PoolBenchmarkHelper},
+	investments::{Investment, InvestmentAccountant, InvestmentProperties, OrderManager},
+};
+use cfg_types::orders::FulfillmentWithPrice;
+use frame_benchmarking::{account, impl_benchmark_test_suite, v2::*, whitelisted_caller};
+use frame_support::traits::fungibles::Mutate;
+use frame_system::RawOrigin;
+use sp_runtime::{traits::One, Perquintill};
+
+use crate::{Call, Config, CurrencyOf, Pallet};
+
+struct Helper<T>(sp_std::marker::PhantomData<T>);
+impl<T: Config> Helper<T>
+where
+	<T::Accountant as InvestmentAccountant<T::AccountId>>::InvestmentInfo:
+		InvestmentProperties<T::AccountId, Currency = CurrencyOf<T>>,
+	T::InvestmentId: Default + Into<CurrencyOf<T>>,
+	T::Accountant: PoolBenchmarkHelper<AccountId = T::AccountId>
+		+ InvestmentIdBenchmarkHelper<
+			InvestmentId = T::InvestmentId,
+			PoolId = <T::Accountant as PoolBenchmarkHelper>::PoolId,
+		>,
+	<T::Accountant as PoolBenchmarkHelper>::PoolId: Default + Copy,
+{
+	fn get_investment_id() -> T::InvestmentId
+where {
+		let pool_id = Default::default();
+		let pool_admin = account("pool_admin", 0, 0);
+
+		T::Accountant::bench_create_pool(pool_id, &pool_admin);
+		T::Accountant::bench_default_investment_id(pool_id)
+	}
+}
+
+#[benchmarks(
+	where
+		<T::Accountant as InvestmentAccountant<T::AccountId>>::InvestmentInfo:
+			InvestmentProperties<T::AccountId, Currency = CurrencyOf<T>>,
+		T::InvestmentId: Default + Into<CurrencyOf<T>>,
+		T::Accountant: PoolBenchmarkHelper<AccountId = T::AccountId>
+			+ InvestmentIdBenchmarkHelper<
+				InvestmentId = T::InvestmentId,
+				PoolId = <T::Accountant as PoolBenchmarkHelper>::PoolId,
+			>,
+		<T::Accountant as PoolBenchmarkHelper>::PoolId: Default + Copy,
+)]
+mod benchmarks {
+	use super::*;
+
+	#[benchmark]
```

Comment:
Just wanted to express that writing v2 benchmarks is definitely more enjoyable to me. Awesome that we can make use of `?`. Hope you feel the same 🙃

Reply:
Yes! When I realized I could remove the `.unwrap()` everywhere it was awesome!

My only concern is what I wrote in the `eng-rust` channel.